oak/server/oak_node.h

```diff
@@ -31,19 +31,21 @@
 
 namespace oak {
 
-typedef std::unordered_map<Handle, std::unique_ptr<ChannelHalf>> ChannelHalfTable;
+using ChannelHalfTable = std::unordered_map<Handle, std::unique_ptr<ChannelHalf>>;
 
 class OakNode final : public NodeThread {
  public:
   // Creates an Oak node by loading the Wasm module code.
-  static std::unique_ptr<OakNode> Create(const std::string& module);
+  static std::unique_ptr<OakNode> Create(const std::string& name, const std::string& module);
```

Comment:
I assume the strings pointed to by `name` and `module` need to outlive this class?

Reply:
I don't think so &ndash; `name` gets copy-constructed into a new `std::string` and `module` ends up at `wabt::ReadBinaryInterp()` which (I think) copies out the relevant sections.